Define and activate Azure 12.0.1 platform release (#412)
giantswarm/releases: azure/v12.0.1/README.md

## :zap:  Giant Swarm Release 12.0.1 for Azure :zap:

**If you are upgrading from 12.0.0, upgrading to this release will not roll your nodes. It will only update the apps.**

> This release upgrades external-dns app to v0.7.3.

Author: ```suggestion
This release upgrades external-dns app to v1.3.0.
```

**Notes for future 12.x.x releases:**

> In order to proceed with upgrade, clusters must be using 11.4.0 or newer release version, otherwise the upgrade request will fail.

Reviewer: What is this about @sslavic can we provide context?

Author: Azure 12+ has nginx optional, but requires that preparatory migration (change of nginx LoadBalancer Service ownership from azure-operator to nginx IC app) work is applied first, there are manual steps, as documented in 11.4.0 release notes. There's even automation in place, admission controller on Azure, ensuring that one cannot skip over multiple major or minor releases. So customer cannot upgrade from say 11.3.3 to 12.0.1, has to upgrade to 11.4.0 first. And this is just note/reminder copied from 12.0.0 release notes. Not sure if we have to preserve it until all customer clusters are upgraded to Azure 12+. It was easier to add, and remove if not needed.

@giantswarm/team-celestial what say you, should we keep this note or rely that SEs and customers read it in 12.0.0 release notes?

Author: Will remove it, automation prevents skipping multiple major or minor releases

> Pay attention to the increased failure rate of requests during upgrade of the master node and include this information in release notes if the problem persists.

Reviewer: ```suggestion
Pay attention to the increased failure rate of requests during upgrade of the master node and include this information in release notes if the problem persists. For context, see [v12.0.0 release notes](https://github.com/giantswarm/releases/tree/master/azure/v12.0.0).
```

Reviewer: @sslavic thank you! Makes sense. Just wanted to provide more context to the reader. Maybe linking to the v12.0.0 releases notes is enough?

Author: Will remove it since there's automation in place which will prevent the upgrade to skip multiple major or minor releases.

> **Notes for future 12.x.x releases:**
> In order to proceed with upgrade, clusters must be using 11.4.0 or newer release version, otherwise the upgrade request will fail.
> Pay attention to the increased failure rate of requests during upgrade of the master node and include this information in release notes if the problem persists.
> 

Author: ```suggestion
```

## :zap:  Giant Swarm Release 12.0.1 for Azure :zap:

**If you are upgrading from 12.0.0, upgrading to this release will not roll your nodes. It will only update the apps.**

This release upgrades external-dns app to v0.7.3.

> 
> **Important Warning**
> During master upgrade from 11.4.0 to 12.0.x, within the time frame of 30 seconds we had noticed a spike in requests failures. This is most likely caused by Azure CNI upgrade and despite our great efforts, we had not found a solution to maintain upgrade path and avoid this disturbance. Please keep this in mind when scheduling an upgrade window, and contact your SE if you have further questions.

Author: ```suggestion
```

Keep just lower warning/note.

## :zap:  Giant Swarm Release 12.0.1 for Azure :zap:

**If you are upgrading from 12.0.0, upgrading to this release will not roll your nodes. It will only update the apps.**

This release upgrades external-dns app to v0.7.3.

> **Important Warning**

Reviewer: @sslavic and maybe @giantswarm/team-celestial , is the Important Warning section a duplicate of Notes for future 12.x.x release? I fear it is two warnings about the same thing? I could be wrong.

Author: I copied this stuff from 12.0.0 release notes

They do overlap to some extent, related to the warning that during upgrade from 11.4.0 to 12.0.x there could be requests failure (downtime) observed. Although they express it a bit differently - maybe rationale for the two variants of the message with different level of details is that one is meant for customer while second one for us publishing new releases.

There's also some uniqueness, e.g. lower one has warning that to upgrade to 12.x one has to upgrade to 11.4.0 first.

We can keep just lower one. But I'm fine in dropping both too.

Author: Kept just "Note for future 12.x.x release", PTAL

During master upgrade from 11.4.0 to 12.0.x, within the time frame of 30 seconds we had noticed a spike in requests failures. This is most likely caused by Azure CNI upgrade and despite our great efforts, we had not found a solution to maintain upgrade path and avoid this disturbance. Please keep this in mind when scheduling an upgrade window, and contact your SE if you have further questions.

> Note that as of 12.0.0 release **NGINX Ingress Controller App** is **optional and not pre-installed** component on Azure. You can find more info in [12.0.0 release notes](https://github.com/giantswarm/releases/tree/master/azure/v12.0.0).

Reviewer: @sslavic do we need this here? It seems out of place with the rest of the content for this release. Also, we mention it in the optional apps announcements.

Author: Removed. PTAL

> 
> Note that as of 12.0.0 release **NGINX Ingress Controller App** is **optional and not pre-installed** component on Azure. You can find more info in [12.0.0 release notes](https://github.com/giantswarm/releases/tree/master/azure/v12.0.0).

Author: ```suggestion
```
